### ISSUE TITLE
usePrismic is undefined when deploying on Netlify

### ISSUE THREAD

**Issue by itzaks** [nuxt-community/prismic-module#157]
### Versions

- @nuxtjs/prismic: 3.0.0-alpha-1

### Issue

Publishing on Netlify with minimal config I'm getting this:

> Cannot destructure property 'client' of 'usePrismic(...)' as it is undefined.

My package.json build script looks like this

```json
"build": "nuxi build",
````

And my `nuxt.config.ts` looks like this:

```ts
  modules: [
    '@nuxtjs/prismic',
  ],
  
  prismic: {
    endpoint: 'xxxxx',
  },
````

---------

**UPDATE**

Even after creating my own composable and using it like this:
````ts
import { createPrismic } from "@prismicio/vue"

export const usePrismicApp = () => {
  const client = createPrismic({
    endpoint: 'xxxx',
    injectComponents: false // Handled at module level
  })

  return client
}
````

I get a similar error
````
Cannot destructure property 'options' of 'usePrismic(...)' as it is undefined.
````

Server logs on Netlify looks like this:

````
Mar 17, 06:15:39 PM: 31273aba WARN   [Vue warn]: inject() can only be used inside setup() or functional components.
Mar 17, 06:15:39 PM: 31273aba ERROR  TypeError: Cannot destructure property 'options' of 'usePrismic(...)' as it is undefined.
    at setup (/var/task/.netlify/functions-internal/server/server.js:75594:17)
    at callWithErrorHandling (/var/task/.netlify/functions-internal/server/server.js:17186:24)
    at setupStatefulComponent (/var/task/.netlify/functions-internal/server/server.js:21400:31)
    at setupComponent2 (/var/task/.netlify/functions-internal/server/server.js:21387:42)
    at renderComponentVNode (/var/task/.netlify/functions-internal/server/server.js:16008:15)
    at Object.ssrRenderComponent (/var/task/.netlify/functions-internal/server/server.js:16368:10)
    at /var/task/.netlify/functions-internal/server/server.js:135662:40
    at renderComponentSubTree (/var/task/.netlify/functions-internal/server/server.js:16062:7)
    at /var/task/.netlify/functions-internal/server/server.js:16017:26
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
````

I'm guessing the above 👆 correlates to how `inject` is used here:
````js
const usePrismic = () => {
  return inject(prismicKey, { options: { endpoint: "" } });
};
````

**Comment by itzaks** [comment]
Creating a fresh repo with minimal setup and only 
````js
 "@nuxtjs/prismic": "alpha",   
 "nuxt3": "latest",
````
as `devDependenies` it seems to work. I'll try to track down what caused this. 

Thanks for a great work with the module.

**Comment by itzaks** [comment]
The error was caused by trying to run the prismic slice machine together with nuxtjs/prismic 3.0. This seems to still not be fully supported. Tracking here instead, [prismicio/slice-machine#420](https://github.com/prismicio/slice-machine/issues/420).

**Comment by lihbr** [comment]
Indeed, Slice Machine doesn't support Nuxt 3 in any way for now and this will likely remain the same for as long as Nuxt 3 is in beta/rc state. Thanks for your contribution!

**Comment by itzaks** [comment]
> Indeed, Slice Machine doesn't support Nuxt 3 in any way for now and this will likely remain the same for as long as Nuxt 3 is in beta/rc state. Thanks for your contribution!

It's strange though, because the error keeps coming up again, without any major changes from my end.  From one commit to the next (only changing some minor html stuff) it comes up. And this is completely without slice-machine. This is my full package.json


````json
{
  "private": true,
  "scripts": {
    "dev": "nuxi dev",
    "build": "nuxi build",
    "start": "node .output/server/index.mjs"
  },
  "devDependencies": {
    "@iconify-json/mdi": "^1.1.2",
    "@nuxtjs/prismic": "alpha",
    "@pinia/nuxt": "^0.1.8",
    "@prismicio/types": "^0.1.27",
    "@types/lodash": "^4.14.180",
    "@unocss/extractor-pug": "^0.29.3",
    "@unocss/nuxt": "^0.29.3",
    "@unocss/preset-icons": "^0.29.3",
    "@unocss/reset": "^0.29.3",
    "@vueuse/nuxt": "^8.1.1",
    "case-it": "^1.0.1",
    "lodash": "^4.17.21",
    "nuxt3": "latest",
    "pinia": "^2.0.12",
    "pinia-plugin-persistedstate": "^1.5.1",
    "ts-debounce": "^4.0.0",
    "unocss": "^0.29.3",
    "vite-plugin-pug": "^0.3.1"
  }
}
````

**Comment by lihbr** [comment]
Thanks for the details!

Definitely, both Nuxt 3 and our Nuxt 3 integration are beta/alpha-ish. I'll keep an eye on those though and make sure the Prismic integration does its best to stay in sync with Nuxt 3 progress :relaxed:

**Comment by itzaks** [comment]
The problem seems to be part of the [prismicio/vue@v3](https://github.com/prismicio/prismic-vue) rather than the Nuxt package upon further investigation.

**Comment by cfab** [comment]
Any news on this ?

**Comment by lihbr** [comment]
Hey there, I'll have a deeper look at it next week, didn't have time earlier.

In the meantime make sure you're running the latest release of Nuxt 3 and the Prismic module. Some changes were released last week on the module following some Nuxt 3 updates.

**Comment by cfab** [comment]
Ok thank you (btw I'm using the latest versions of both nuxt and prismic module).

**Comment by lihbr** [comment]
Hey there,

First of all, sorry for the delay it took to get back to you. I'll make sure to ramp up our Nuxt 3 support effort as the framework leans toward its full release.

I just released a patch fixing the primary issue (`usePrismic()` is `undefined`) when `@nuxtjs/prismic` is installed as a regular dependency (basically the fix was forcing Nuxt to transpile our Vue plugin, see: ead4eb1071e47c6e175bc557003f4fef22e214cb) 

_Install it with `npm install @nuxtjs/prismic@3.0.0-alpha.3`_

Overall I'd recommend keeping `@nuxtjs/prismic` as a dev dependency. However, if having it as a regular dependency can be an escape hatch it's fine to keep it as it (I guess? I asked Nuxt team about guidelines on that matter).

I'll close this issue as everything related directly to `@nuxtjs/prismic` seem to have been fixed. We'll track progress about the Slice Machine related issue on: https://github.com/prismicio/slice-machine/issues/420 (and I'll ping Slice Machine team on it!)

Thanks for your contribution! 💚
